File: robot-return-to-origin-657/Solution.java
/**
 * LeetCode problem # 657, 'Robot Return to Origin'
 * https://leetcode.com/problems/robot-return-to-origin
 */

class Solution {
    public boolean judgeCircle(String moves) {
        int row = 0;
        int col = 0;

        for (int i = 0; i < moves.length(); ++i) {
            char move = moves.charAt(i);
            switch (move) {
            case 'U':
                --col;
                break;
            case 'D':
                ++col;
                break;
            case 'L':
                --row;
                break;
            case 'R':
                ++row;
                break;
            default:
                break;
            }
        }
        return row == 0 && col == 0;
    }
}

File: robot-return-to-origin-657/solution.js
/**
 * LeetCode problem # 657, 'Robot Return to Origin'
 * https://leetcode.com/problems/robot-return-to-origin
 */

/**
 * @param {string} moves
 * @return {boolean}
 */
var judgeCircle = function (moves) {
  const fr = { U: 0, D: 0, L: 0, R: 0 };

  for (const move of moves) {
    fr[move] = (fr[move] ?? 0) + 1;
  }

  return fr["D"] === fr["U"] && fr["L"] === fr["R"];
};
